### PR TITLE
unix: suppress EINTR/EINPROGRESS in uv_fs_close()

### PR DESCRIPTION
**Pull request by bnoordhuis**
close() is interruptible, meaning that a signal delivered when the
close() system call is executing, can result in a return value of
-1 with errno set to EINTR or EINPROGRESS. A survey of supported
Unices suggests the file descriptor is in fact closed and so there
is nothing to do except ignore the error.

Libuv already handled it correctly for internal file descriptors
throug uv__close() and uv__close_nocheckstdio() but uv_fs_close()
did not until now.

CI: https://ci.nodejs.org/job/libuv-test-commit/1331/ (2 expected failures until #2234 lands)

### CODE REVIEW

File: src/unix/fs.c [hunk 1]
```diff
@@ -143,6 +143,18 @@ extern char *mkdtemp(char *template); /* See issue #740 on AIX < 7 */
   while (0)
 
 
+static int uv__fs_close(int fd) {
+  int rc;
+
+  rc = close(fd);
+  if (rc == -1)
```
Comment (cjihrig) [inline src/unix/fs.c]: Any reason not to combine the two `if` statements?

**Comment by cjihrig** [comment]
CI was good.

File: src/unix/fs.c [hunk 2]
```diff
@@ -143,6 +143,18 @@ extern char *mkdtemp(char *template); /* See issue #740 on AIX < 7 */
...
+    if (errno == EINTR || errno == EINPROGRESS)
+      rc = 0;  /* The close is in progress, not an error. */
+
+  return rc;
```
Comment (santigimeno) [inline src/unix/fs.c]: Use `UV__ERR()` in case of error?

File: src/unix/fs.c (hunk 1 above)
Comment (bnoordhuis) [reply src/unix/fs.c]: Personal preference, really. I find two if statements easier to read than `(c1 && (c2 || c3))`.

File: src/unix/fs.c (hunk 2 above)
Comment (bnoordhuis) [reply src/unix/fs.c]: This is called by `uv__fs_work()` which looks at the return code and stores `UV__ERR(errno)` when it's -1.

Comment (santigimeno) [reply src/unix/fs.c]: :+1:  Right.

**Comment by cjihrig** [comment]
Fresh CI with no related failures: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/1354/

Landed in https://github.com/libuv/libuv/commit/a40a5e897d508091814a3aab19f0fa673ea52fce. Thanks!

**Pull request closed without merging**